File: 1127-last-stone-weight/last-stone-weight.py
class Solution(object):
    def lastStoneWeight(self, stones):
        """
        :type stones: List[int]
        :rtype: int
        """
        stones.sort()
        while len(stones)>1:
            stone1 = stones.pop()
            stone2 = stones.pop()
            if stone1 != stone2:
                stones.append(stone1-stone2)
                stones.sort()
        return stones[0] if stones else 0
        

File: 1127-last-stone-weight/last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> pq(stones.begin(), stones.end());
        // Makes it a maxheap

        while(pq.size()>1){
            int stone1 = pq.top();
            pq.pop();
            int stone2 = pq.top();
            pq.pop();

            if(stone1 != stone2){
                pq.push(abs(stone1-stone2));
            }

           

        }
         return pq.empty() ? 0 : pq.top();
    }
    
        
    
};